starboard/android/shared/application_android.cc

```diff
@@ -242,6 +246,11 @@ void ApplicationAndroid::WakeSystemEventWait() {
   ALooper_wake(looper_);
 }
 
+void ApplicationAndroid::Inject(Event* event) {
+  QueueApplication::Inject(event);
+  WakeSystemEventWait();
```

Comment:
QueueApplication::Inject(event); can conditionally run into WakeSystemEventWait(); in its implementation, how does this fix the problem?

Reply:
This doesn't fix the problem, but avoids waiting for `kMaxPollingTimeMillisecond` (10ms) when an event is injected.

Reply:
when handle_system_events_ is true, it calls WakeSystemEventWait(). If we want to unconditionally call it, should we replace MayHaveSystemEvents() in application_android.h to
```
bool MayHaveSystemEvents() override { return true; }
```
Reading the comments on top of where handle_system_events_ is set, there is the reason why it existed there?

Reply:
Ahh, right, and `MayHaveSystemEvents` now returns a non-threadsafe bool. Fixed.